LBHackney-IT/lbh-social-care
# Added ObjectInput

**What**  
- Added ObjectInput form component, it's also supporting inline formatting (with `isInline` prop)
- Extended Summary to show object components inline, using `summaryInline`
- Added _object step_ in the test form, for demo purpose of the feature

**Example:**
The component is going to be declared as follow:

```
{
  component: 'Object',
  name: 'phone_number',
  label: 'Phone Number',
  components: [
    {
      component: 'PhoneInput',
      name: 'phoneNumber',
      label: 'Phone number',
      rules: { required: true },
    },
    {
      component: 'TextInput',
      name: 'phoneType',
      label: 'Phone type',
    },
  ],
},
```
and it's returning:
```
{
  phoneType: "main contact",
  phoneNumber: "077..."
}
```

<img width="1004" alt="Screenshot 2021-01-22 at 18 09 32" src="https://user-images.githubusercontent.com/435399/105576517-7f1a8a00-5d73-11eb-861b-0edc5853e668.png">
<img width="1004" alt="Screenshot 2021-01-22 at 18 19 11" src="https://user-images.githubusercontent.com/435399/105576518-7fb32080-5d73-11eb-9532-7e90334b18b1.png">
<img width="1006" alt="Screenshot 2021-01-22 at 18 21 10" src="https://user-images.githubusercontent.com/435399/105576520-804bb700-5d73-11eb-9cca-70061a70cb52.png">

**Extras**
- Moved `Button` component outside the `Form` as it's used in several different places, unrelated to the Forms
- Moved `DynamicStep` and `DynamicInput` inside the `FormWizard` folder

**How to test it**
- go to http://dev.hackney.gov.uk:3000/form/test/first-step
- select **show object step**
- check that is working properly
